### ISSUE TITLE
llvm function multi-versioning

### ISSUE THREAD

**Issue by enh-google** [google/android-riscv64#69]
related to but distinct from https://github.com/google/android-riscv64/issues/8, we'll want function multi-versioning in llvm.

**Comment by MaoHan001** [comment]
Is this issue related to the multi-versioning feature itself or to function optimization that supports different instruction set extensions?

**Comment by nick-knight** [comment]
> ifuncs are too much trouble [...] fmv will help

To the best of my knowledge, function multiversioning in both LLVM and GCC is implemented using GNU indirect functions. So this ticket could be considered blocked on #8.

**Comment by kito-cheng** [comment]
This is referring something like this? https://gcc.gnu.org/wiki/FunctionMultiVersioning

**Comment by nick-knight** [comment]
Yes. As usual,  @maskray's blog has useful info: https://maskray.me/blog/2023-02-05-function-multi-versioning

**Comment by topperc** [comment]
Kito filed an issue here https://github.com/riscv-non-isa/riscv-c-api-doc/pull/35
